test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -2195,18 +2230,31 @@ TEST_P(LdsTest, ChooseMatchedDomain) {
       ->mutable_routes(0)
       ->mutable_route()
       ->mutable_cluster_header();
-  balancers_[0]->ads_service()->SetLdsResource(
-      AdsServiceImpl::BuildListener(route_config), kDefaultResourceName);
+  if (GetParam().enable_rds_testing()) {
+    balancers_[0]->ads_service()->SetRdsResource(route_config,
+                                                 kDefaultResourceName);
+  } else {
+    balancers_[0]->ads_service()->SetLdsResource(
+        AdsServiceImpl::BuildListener(route_config), kDefaultResourceName);
+  }
   SetNextResolution({});
   SetNextResolutionForLbChannelAllBalancers();
   (void)SendRpc();
-  EXPECT_EQ(balancers_[0]->ads_service()->lds_response_state(),
-            AdsServiceImpl::ACKED);
+  if (GetParam().enable_rds_testing()) {
```

Comment:
Similarly, I suggest creating a helper function in `XdsEnd2endTest` for this:

```
AdsServiceImpl::ResponseState RouteConfigurationResponseState(int idx) const {
  AdsServiceImpl* ads_service = balancers_[idx]->ads_service();
  if (GetParam().enable_rds_testing()) {
    return ads_service->rds_response_state();
  }
  return ads_service->lds_response_state();
}
```

Then the code in each test can just say something like this:

```
EXPECT_EQ(RouteConfigurationResponseState(0), AdsServiceImpl::ACKED);
```